src/PgRvn.Server/Session.cs

```diff
@@ -0,0 +1,200 @@
+﻿using Raven.Client.Documents;
+using System;
+using System.Collections.Generic;
+using System.IO.Pipelines;
+using System.Net.Sockets;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace PgRvn.Server
+{
+    public class PgFatalException : Exception
+    {
+        public string ErrorCode;
+
+        /// <summary>
+        /// Creates an Postgres exception to be sent back to the client
+        /// </summary>
+        /// <param name="errorCode">A Postgres error code (SqlState). See <see cref="PgErrorCodes"/></param>
+        /// <param name="errorMessage">Error message</param>
+        /// <returns>ErrorResponse message</returns>
+        public PgFatalException(string errorCode, string errorMessage) : base(errorMessage)
+        {
+            ErrorCode = errorCode;
+        }
+    }
+
+    public class PgErrorException : Exception
```

Comment:
Move exceptions to dedicated classes (e.g. in `Exceptions` subdirectory)

Reply:
Done